pkg/router/controller/unique_host_test.go

```diff
@@ -67,6 +76,24 @@ func TestValidateHostName(t *testing.T) {
 			},
 			expectedErrors: true,
 		},
+		{
+			name: "invalid-name-one-label",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					Host: "org",
+				},
+			},
+			expectedErrors: true,
+		},
+		{
+			name: "invalid-name-asterisk",
+			route: &routev1.Route{
+				Spec: routev1.RouteSpec{
+					Host: "*",
+				},
+			},
+			expectedErrors: true,
+		},
```

Comment:
Might as well add a couple test cases with the annotation.

Reply:
I can add some. There are some test cases in the complement PR: https://github.com/openshift/openshift-apiserver/pull/167/files#diff-e571a492e8cca2251f9208e5274c4c9a4e67368113b1b6f131dd5b685c5eaa56

Reply:
It'd be good to add a test case or two here to get coverage to make sure `ValidateHostName` correctly looks up the annotation and passes it to `ValidateHost`.